vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceOverviewJsonFormat.java

```diff
@@ -0,0 +1,190 @@
+import edu.asu.diging.vspace.web.staff.SpaceController;
+
+@Component
+public class SpaceOverviewJsonFormat {
+
+    @Autowired
+    private SpaceRepository spaceRepo;
+
+    @Autowired
+    private ModuleRepository moduleRepo;
+
+    /**
+     * creating list of SpaceOverview object for space nodes in the spaceoverview
+     * graph
+     * 
+     * @param contextPath   This variable holds the contextpath of the application
+     * @param SpaceNodeList List of spaces
+     * @throws JsonProcessingException
+     */
+    private List<SpaceOverview> constructNodesForSpaces(String contextPath, Iterable<Space> spaceNodeList)
+            throws JsonProcessingException {
+
+        List<SpaceOverview> spaceVertexList = new ArrayList<>();
+        if (spaceNodeList != null) {
+            StringBuilder linkPathBuilder = new StringBuilder();
+            StringBuilder imagePathBuilder = new StringBuilder();
+            for (Space spaceNode : spaceNodeList) {
+                SpaceOverview spaceOverview = createSpaceNode(linkPathBuilder, imagePathBuilder, spaceNode,
+                        contextPath);
+                spaceVertexList.add(spaceOverview);
+            }
+        }
+        return spaceVertexList;
+    }
+
+    /**
+     * creating list of SpaceOverview object for module nodes in the spaceoverview
+     * graph
+     * 
+     * @param contextPath    This variable holds the contextpath of the application
+     * @param moduleNodeList List of modules
+     * @throws JsonProcessingException
+     */
+    private List<SpaceOverview> constructNodesForModules(String contextPath, Iterable<Module> moduleNodeList)
+            throws JsonProcessingException {
+
+        List<SpaceOverview> moduleVertexList = new ArrayList<>();
+
+        if (moduleNodeList != null) {
+            StringBuilder linkPathBuilder = new StringBuilder();
+            StringBuilder imagePathBuilder = new StringBuilder();
+            for (Module moduleNode : moduleNodeList) {
+                SpaceOverview spaceOverview = createModuleNode(linkPathBuilder, imagePathBuilder, moduleNode,
+                        contextPath);
+                moduleVertexList.add(spaceOverview);
+            }
+        }
+        return moduleVertexList;
+    }
+
+    /**
+     * Populating space node in the spaceoverview graph
+     * 
+     * @param linkPathBuilder  StringBuilder object to create the URL link for space
+     *                         in staff side
+     * @param imagePathBuilder StringBuilder object to create the URL link for image
+     *                         in staff side
+     * @param space            passing the space object
+     * 
+     * @param contextPath      This variable holds the contextpath of the
+     *                         application
+     */
+    private SpaceOverview createSpaceNode(StringBuilder linkPathBuilder, StringBuilder imagePathBuilder, Space space,
+            String contextPath) {
```

Comment:
why are the string builders being passed here? they don't seem to serve a purpose outside of this method.